File: minhas-financas-front-end/src/types/resumo-financeiro-ia.ts
import {
  IndicadorFinanceiroSaude,
  PainelIndicadoresFinanceirosSaude,
  ResumoSaudeFinanceira,
} from "@/types/saude-financeira";

export interface InsightFinanceiroData {
  codigoIndicadorRelacionado?: number | null;
  tipo: number;
  prioridade: number;
  titulo: string;
  descricao: string;
  acaoSugerida: string;
}

export interface PainelInsightsFinanceirosData {
  todos: InsightFinanceiroData[];
  prioritarios: InsightFinanceiroData[];
  destaquesPositivos: InsightFinanceiroData[];
}

export interface ResumoFinanceiroIAData {
  dataReferencia: string;
  saudeFinanceira: ResumoSaudeFinanceira;
  indicadores: PainelIndicadoresFinanceirosSaude;
  insights: PainelInsightsFinanceirosData;
  resumoExecutivo: string;
  prioridadesImediatas: string[];
  destaquesPositivos: string[];
}

export type IndicadorResumoFinanceiroIA = IndicadorFinanceiroSaude;

File: minhas-financas-back-end/MinhasFinancas.Domain/Services/AnaliseFinanceira/Modelos/ResumoFinanceiroIA.cs
namespace MinhasFinancas.Domain.Services.AnaliseFinanceira.Modelos
{
    public class ResumoFinanceiroIA
    {
        public DateTime DataReferencia { get; set; }
        public ResumoSaudeFinanceira SaudeFinanceira { get; set; } = new();
        public PainelIndicadoresFinanceiros Indicadores { get; set; } = new();
        public PainelInsightsFinanceiros Insights { get; set; } = new();
        public string ResumoExecutivo { get; set; } = string.Empty;
        public List<string> PrioridadesImediatas { get; set; } = [];
        public List<string> DestaquesPositivos { get; set; } = [];
    }
}
